examples/water/main.rs

```diff
@@ -0,0 +1,797 @@
+                stencil_back: wgpu::StencilStateFaceDescriptor::IGNORE,
+                stencil_read_mask: 0,
+                stencil_write_mask: 0,
+            }),
+            vertex_state: wgpu::VertexStateDescriptor {
+                index_format: wgpu::IndexFormat::Uint16,
+                vertex_buffers: &[
+                    wgpu::VertexBufferDescriptor {
+                        stride: terrain_vertex_size as wgpu::BufferAddress,
+                        step_mode: wgpu::InputStepMode::Vertex,
+                        attributes: &[
+                            wgpu::VertexAttributeDescriptor { //Position
+                                format: wgpu::VertexFormat::Float3,
+                                offset: 0,
+                                shader_location: 0,
+                            },
+                            wgpu::VertexAttributeDescriptor { //Normal
+                                format: wgpu::VertexFormat::Float3,
+                                offset: std::mem::size_of::<[f32; 3]>() as wgpu::BufferAddress,
+                                shader_location: 1,
+                            },
+                            wgpu::VertexAttributeDescriptor { //Colour
+                                format: wgpu::VertexFormat::Uchar4Norm,
+                                offset: std::mem::size_of::<[f32; 3]>() as wgpu::BufferAddress * 2,
+                                shader_location: 2
+                            },
+                        ],
+                    },
+                ],
+            },
+            sample_count: 1,
+            sample_mask: !0,
+            alpha_to_coverage_enabled: false,
+        });
+
+        // Done
+        let this = Example {
+            water_vertex_buf,
+            water_vertex_count: water_vertices.len(),
+            water_bind_group,
+            water_uniform_buf,
+            water_pipeline,
+
+            terrain_vertex_buf,
+            terrain_vertex_count: terrain_vertices.len(),
+            terrain_normal_bind_group,
+            terrain_flipped_bind_group,
+            terrain_normal_uniform_buf,
+            terrain_flipped_uniform_buf,
+            terrain_pipeline,
+
+            reflect_view: reflection_texture.create_default_view(),
+
+            draw_depth_buffer: draw_depth_buffer.create_default_view(),
+
+            current_frame: 0,
+
+            active: 0,
+        };
+        (this, None)
+    }
+
+    fn update(&mut self, _event: winit::event::WindowEvent) {
+        //empty
+    }
+
+    fn resize(
+        &mut self,
+        sc_desc: &wgpu::SwapChainDescriptor,
+        device: &wgpu::Device,
+        queue: &wgpu::Queue,
+    ) {
+        if sc_desc.width == 0 && sc_desc.height == 0 {
+            // Stop rendering altogether.
+            self.active = std::usize::MAX;
+            return;
+        } else {
+            // Skip a frame, since it crashes otherwise.
```

Comment:
what is this crash about?

Reply:
The `sc_desc` passed in the frame we undo minimization has the correct size (1024 x 768), however the next call to render has frames whose sizes are 0 x 0:
```rust
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `Extent { width: 1024, height: 768, depth: 1 }`,
 right: `Extent { width: 0, height: 0, depth: 1 }`: Extent state must match extent from view'
```
Skipping a frame seemed like a valid way of circumventing this.

Reply:
ok, I see. I suppose it's related to the resizing bugs of examples we have open :sweat_smile: